glowfic-dl.py

```diff
@@ -207,13 +308,30 @@ async def download_chapter(
     return (title, list(render_posts(posts, image_map, authors)))
 
 
-def compile_chapters(chapters: list[tuple[str, list[Section]]]) -> list[epub.EpubHtml]:
+def compile_chapters(
+    chapters: list[tuple[str, list[Section]]]
+) -> Iterable[list[epub.EpubHtml]]:
+    chapter_digits = len(str(len(chapters)))
     anchor_sections = {}
+
+    # Map permalinks to file names
     for (i, (title, sections)) in enumerate(chapters):
+        section_digits = len(str(len(sections)))
         for (j, section) in enumerate(sections):
-            file_name = "chapter%i_%i.html" % (i, j)
+            file_name = "Text/" + make_filename_valid_for_epub3(
+                "%.*i-%.*i (%s).xhtml"
+                % (
+                    chapter_digits,
+                    i + 1,
+                    section_digits,
+                    j,
+                    title,
+                )
+            )
             for permalink in section.link_targets:
                 anchor_sections[permalink] = file_name
+
+    # Replace external links with internal links where practical
```

Comment:
Nit: Maybe "possible"? "Practicial" to me implies that there are some links we could convert in theory but don't because it's too hard.

Reply:
I've updated the phrasing. And also discovered it's not actually accurate, currently; apparently some link-replacement actually fails, currently. (See the [planecrash lectures](https://glowfic.com/board_sections/721) section, for a lightweight example: the link at the end of `to hell with science`, in the downloaded EPUB, goes out to glowfic.com rather than going to the book's internal `to earth with science` copy.) So I'll add that to my pile of things to try to fix in my next PR.